pkg/provisioner/controller.go

```diff
@@ -200,28 +200,23 @@ func (c *obcController) processNextItemInQueue() bool {
 func (c *obcController) syncHandler(key string) error {
 
 	setLoggersWithRequest(key)
-	logD.Info("new Reconcile iteration")
+	logD.Info("reconciling claim")
 
 	obc, err := claimForKey(key, c.libClientset)
 	if err != nil {
-		return fmt.Errorf("request key %q: %v", key, err)
-	}
-	if obc == nil {
-		return fmt.Errorf("unexpected nil obc for request key %q", key)
+		if errors.IsNotFound(err) {
+			log.Info("OBC vanished, assuming it was deleted")
+			return nil
+		}
+		return fmt.Errorf("could not sync OBC %s: %v", key, err)
 	}
 
-	deleteEvent := obc.ObjectMeta.DeletionTimestamp != nil
-
-	if deleteEvent {
-		// ***********************
-		// Delete or Revoke Bucket
-		// ***********************
+	// ***********************
+	// Delete or Revoke Bucket
+	// ***********************
+	if obc.ObjectMeta.DeletionTimestamp != nil {
 		log.Info("OBC deleted, proceeding with cleanup")
-		err = c.handleDeleteClaim(key, obc)
-		if err != nil {
-			log.Error(err, "error cleaning up OBC", "name", key)
-		}
-		return err
+		return c.handleDeleteClaim(key, obc)
```

Comment:
In the `updateObjectBucketClaimPhase` call below _(sorry I don't know how to add a comment to line that wasn't changed)_ why do we update the phase before we know if the provisioner name matches? It's possible, isn't it, that this obc is for a different provisioner and we've just updated it?

Reply:
+100, no idea if or when that moved. will fix

Reply:
fixed